### PR TITLE
Feat/met 4736 view harvesting parameters

### PR DESCRIPTION
**Pull request by JoanaCMS**

### CODE REVIEW

File: src/main/java/eu/europeana/metis/sandbox/common/Protocol.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package eu.europeana.metis.sandbox.common;
+
+/**
+ * Enum class to represent the type of harvesting
+ */
+public enum Protocol {
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/common/Protocol.java:6]: Sound good, just renamed

File: src/main/java/eu/europeana/metis/sandbox/dto/FileHarvestingDto.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * Represents the data related to File harvesting
+ */
+@ApiModel(FileHarvestingDto.SWAGGER_MODEL_NAME)
+public class FileHarvestingDto implements HarvestingParametersDto {
+
+    public static final String SWAGGER_MODEL_NAME = "FileUpload";
+
+    @JsonProperty("protocol")
+    private final Protocol protocol = Protocol.FILE;
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/dto/FileHarvestingDto.java:16]: Just fixed

File: src/main/java/eu/europeana/metis/sandbox/dto/FileHarvestingDto.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * Represents the data related to File harvesting
+ */
+@ApiModel(FileHarvestingDto.SWAGGER_MODEL_NAME)
+public class FileHarvestingDto implements HarvestingParametersDto {
+
+    public static final String SWAGGER_MODEL_NAME = "FileUpload";
+
+    @JsonProperty("protocol")
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/dto/FileHarvestingDto.java:15]: Done!

File: src/main/java/eu/europeana/metis/sandbox/dto/HttpHarvestingDto.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Represents the data related to Http harvesting
+ */
+@ApiModel(HttpHarvestingDto.SWAGGER_MODEL_NAME)
+public class HttpHarvestingDto implements HarvestingParametersDto {
+
+    public static final String SWAGGER_MODEL_NAME = "HttpUpload";
+
+    @JsonProperty("protocol")
+    private final Protocol protocol = Protocol.HTTP;
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/dto/HttpHarvestingDto.java:16]: Done

File: src/main/java/eu/europeana/metis/sandbox/dto/HttpHarvestingDto.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * Represents the data related to Http harvesting
+ */
+@ApiModel(HttpHarvestingDto.SWAGGER_MODEL_NAME)
+public class HttpHarvestingDto implements HarvestingParametersDto {
+
+    public static final String SWAGGER_MODEL_NAME = "HttpUpload";
+
+    @JsonProperty("protocol")
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/dto/HttpHarvestingDto.java:15]: Done

File: src/main/java/eu/europeana/metis/sandbox/dto/OAIPmhHarvestingDto.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * Represents the data related to OAI-PMH harvesting
+ */
+@ApiModel(OAIPmhHarvestingDto.SWAGGER_MODEL_NAME)
+public class OAIPmhHarvestingDto implements HarvestingParametersDto {
+
+    public static final String SWAGGER_MODEL_NAME = "OAIPmhUpload";
+
+    @JsonProperty("protocol")
+    private final Protocol protocol = Protocol.OAI_PMH;
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/dto/OAIPmhHarvestingDto.java:16]: Done

File: src/main/java/eu/europeana/metis/sandbox/dto/OAIPmhHarvestingDto.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * Represents the data related to OAI-PMH harvesting
+ */
+@ApiModel(OAIPmhHarvestingDto.SWAGGER_MODEL_NAME)
+public class OAIPmhHarvestingDto implements HarvestingParametersDto {
+
+    public static final String SWAGGER_MODEL_NAME = "OAIPmhUpload";
+
+    @JsonProperty("protocol")
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/dto/OAIPmhHarvestingDto.java:15]: Done

File: src/main/java/eu/europeana/metis/sandbox/dto/FileHarvestingDto.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import eu.europeana.metis.sandbox.common.Protocol;
+import io.swagger.annotations.ApiModel;
+
+/**
+ * Represents the data related to File harvesting
+ */
+@ApiModel(FileHarvestingDto.SWAGGER_MODEL_NAME)
+public class FileHarvestingDto implements HarvestingParametersDto {
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/dto/FileHarvestingDto.java:11]: I can do that, but I was following the same "structure" from other DTO classes. Should I change for other classes as well?

File: src/main/java/eu/europeana/metis/sandbox/dto/HarvestingParametersDto.java [hunk 9]
```diff
@@ -0,0 +1,7 @@
+package eu.europeana.metis.sandbox.dto;
+
+/**
+ * Interface that represents all type of harvesting with their parameters
+ */
+public interface HarvestingParametersDto {
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/dto/HarvestingParametersDto.java:6]: Done

File: src/main/java/eu/europeana/metis/sandbox/repository/HarvestingParametersRepository.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
+package eu.europeana.metis.sandbox.repository;
+
+import eu.europeana.metis.sandbox.entity.HarvestingParametersEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+/**
+ * Repository connected to Harvesting Parameters table
+ */
+public interface HarvestingParametersRepository extends JpaRepository<HarvestingParametersEntity, Long> {
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/repository/HarvestingParametersRepository.java:9]: Done

File: src/main/resources/database/schema.sql [hunk 11]
```diff
@@ -83,6 +83,21 @@ CREATE TABLE IF NOT EXISTS default_transform_xslt
     PRIMARY KEY (id)
 );
 
+CREATE TABLE IF NOT EXISTS harvesting_parameters
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/resources/database/schema.sql:86]: Done

File: src/main/resources/database/schema.sql [hunk 12]
```diff
@@ -83,6 +83,21 @@ CREATE TABLE IF NOT EXISTS default_transform_xslt
...
+    dataset_id       BIGINT      NOT NULL,
+    protocol         VARCHAR(10) NOT NULL,
+    file_name        TEXT        NULL,
+    file_type        VARCHAR(9)  NULL,
+    url              TEXT        NULL,
+    set_spec         TEXT        NULL,
+    metadata_format  TEXT        NULL,
+    PRIMARY KEY (id),
+    FOREIGN KEY (dataset_id) REFERENCES dataset (dataset_id)
+    );
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/resources/database/schema.sql:98]: Fixed

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/HarvestingParametersServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+    public HarvestingParametersServiceImpl(HarvestingParametersRepository harvestingParametersRepository,
+                                           DatasetRepository datasetRepository) {
+        this.harvestingParametersRepository = harvestingParametersRepository;
+        this.datasetRepository = datasetRepository;
+    }
+
+
+    @Override
+    @Transactional
+    public void createDatasetHarvestingParameters(String datasetId, Protocol protocol,
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/service/dataset/HarvestingParametersServiceImpl.java:37]: Done, tell me what do you think of this implementation

File: src/main/java/eu/europeana/metis/sandbox/dto/FileHarvestingDto.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (jeortizquan) [reply src/main/java/eu/europeana/metis/sandbox/dto/FileHarvestingDto.java]: :thinking: I see, lets leave them like this, for the moment. I was thinking that DTO are initials that's why I was thinking to use in all caps.

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/HarvestingParametersServiceImpl.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (jeortizquan) [reply src/main/java/eu/europeana/metis/sandbox/service/dataset/HarvestingParametersServiceImpl.java]: Yes. it looks good to me.

File: src/main/java/eu/europeana/metis/sandbox/entity/HarvestingParameterEntity.java [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+
+    private String fileType;
+
+    private String url;
+
+    private String setSpec;
+
+    private String metadataFormat;
+
+    /**
```
Comment (jeortizquan) [inline src/main/java/eu/europeana/metis/sandbox/entity/HarvestingParameterEntity.java:47]: remove this constructor so only the interface is used.
```
  /**
     * Parameterized constructor
     * @param datasetId The id of the dataset associated to these harvesting parameters
     * @param harvestProtocol The type of harvesting done
     * @param fileName The name of ile that was harvested (if it exists)
     * @param fileType The type of the file that was harvested (if it exists)
     * @param url The url used to be harvested (if it exists)
     * @param setSpec The setspec used for the harvesting (if it exists)
     * @param metadataFormat The metadata format used for the harvesting (if it exists)
     */
    public HarvestingParameterEntity(DatasetEntity datasetId, HarvestProtocol harvestProtocol, String fileName, String fileType,
                                     String url, String setSpec, String metadataFormat) {
        this.datasetId = datasetId;
        this.harvestProtocol = harvestProtocol;
        this.fileName = fileName;
        this.fileType = fileType;
        this.url = url;
        this.setSpec = setSpec;
        this.metadataFormat = metadataFormat;
    }
```

File: src/main/resources/performance/scenario-1-harvest-file.jmx
Comment (jeortizquan) [inline src/main/resources/performance/scenario-1-harvest-file.jmx:1]: please return the name here to `protocol` here it has another meaning.

File: src/main/resources/performance/scenario-2-harvest-oaipmh.jmx
Comment (jeortizquan) [inline src/main/resources/performance/scenario-2-harvest-oaipmh.jmx:1]: same here please return the name to `protocol` it has another meaning here.

File: src/main/resources/performance/scenario-3-harvest-file-and-oaipmh.jmx
Comment (jeortizquan) [inline src/main/resources/performance/scenario-3-harvest-file-and-oaipmh.jmx:1]: and here please return the name to `protocol` it has another meaning here.

**Review by jeortizquan — CHANGES_REQUESTED** [review]
There was some "find replace" in the .jmx files that should not be changed please return them to `protocol`

File: src/main/java/eu/europeana/metis/sandbox/entity/HarvestingParameterEntity.java (hunk 14 above)
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/entity/HarvestingParameterEntity.java:47]: Done

File: src/main/resources/performance/scenario-2-harvest-oaipmh.jmx
Comment (JoanaCMS) [reply src/main/resources/performance/scenario-2-harvest-oaipmh.jmx:1]: Done

File: src/main/resources/performance/scenario-3-harvest-file-and-oaipmh.jmx
Comment (JoanaCMS) [reply src/main/resources/performance/scenario-3-harvest-file-and-oaipmh.jmx:1]: Done

**Review by jeortizquan — CHANGES_REQUESTED** [review]
Please have a look on the sonarqube code smells.

**Review by jeortizquan — APPROVED** [review]

**Pull request merged**